tests/scripts/analyze_outcomes.py

```diff
@@ -15,25 +15,31 @@
 
 import check_test_cases
 
-class Results:
+class TestLog:
     """Process analysis results."""
 
     def __init__(self):
         self.error_count = 0
         self.warning_count = 0
+        self.output = ""
 
-    @staticmethod
-    def log(fmt, *args, **kwargs):
-        sys.stderr.write((fmt + '\n').format(*args, **kwargs))
+    def add_line(self, fmt, *args, **kwargs):
+        self.output = self.output + (fmt + '\n').format(*args, **kwargs)
```

Comment:
I though that in case of multiple tests run in sequence, it could be helpful to have their output changed all together in the output log file without any other test's output interleaving.